README.MD

```diff
@@ -50,19 +55,78 @@ out of the box configuration for both `config/application-example.yml` and
 `config/database-example.yml`. Small changes that are described below are
 needed for certificates and setup of EPP/REST API.
 
+### Build docker images
+
+In the project root, run the following command:
+
+```bash
+$ docker-compose build
+```
+
+__NB!__ Containers need to be rebuilt every time there is an update to the
+gemfile.
+
+### Fire up docker-compose project
+
+Since multiple configuration files are already using container-based paths, we need to boot up our project beforehand to address these paths.
+
+In the project root, run the following command:
+
+```bash
+$ docker-compose up -d
+```
+
+This command will start all containers defined in docker-compose configuration file. You will see some errors from docker logs, but don't worry, they're expected.
+
+### Connect to active registry container
+
+Since openssl.cnf is based on paths used in container itself, we need to generate certificates inside container environment.
+
+In the project root, run the following command:
+
+```bash
+$ docker ps
+```
+
+Mark down any `CONTAINER ID` which is running `registry` image.
+```
+$ docker ps
+CONTAINER ID        IMAGE
+2b13fa1fe93c        auction_center
+**d4077e3a2e1c** << THIS ONE        registry
+b0934102c7a6        auction_center
+**48e8e38d2059** << OR THIS ONE       registry
+9b6e13696d01        epp_proxy
+7efc7918ecf0        internetee/httpd_epp:2.2
+54a6e24954d6        postgres:9.4
+
+```
+
+Now, let's connect to registry container via:
+
+```bash
+$ docker exec -ti <CONTAINER ID> /bin/bash
+```
+
+Great, we're inside container now.
+
 ### Setup root CA
 
 You will be asked to supply a password and certain certificate details, such as
 common name. Remember the password, as you'll be required to supply it at
 various points in time for generation of other certificates. The root CA, and
 all the certificates are valid for 1 year only.
 
+Now, while connected to `registry` container, run this:
```

Comment:
Why you want to generate the ca inside the container if the shared/ca is included in the container ?

Reply:
I'm not sure if it's just me, but I ran into bunch of errors while trying to generate certificates on local- serial number issues, openssl config path issues, possibly something else as well which I don't remember at this point. Maybe I missed something myself, but based on my experience, generating certs in registry env went without any pain whatsoever. If you have any idea what could have went wrong, then I'm all ears 🙂

Reply:
Agreed with Karl, this helps to simplify gen certs a lot.